### ISSUE TITLE
Train loss = nan after few steps

### ISSUE THREAD

**Issue by Lakshya-Kejriwal** [qqwweee/keras-yolo3#121]
Hi,

I know this issue has been resolved in commit https://github.com/qqwweee/keras-yolo3/commit/b7cbd24d1b201d2d3e10d33fd79d3fd60d21a858. I am trying to train the model on custom data set that consists of single channel images. I modified `utils.py` to handle single channel images. The loss decreases for a few steps but after that it becomes nan and never comes back. Is anyone else still facing this problem?

**Comment by oak-tree** [comment]
check if you get `inf` before the `nan` somwhere. For me the issue was that one of the box somewhere had a dim of zero in one of its axis.  which result with log(0)

**Comment by datduonguva** [comment]
Here is my solution
Hope it would help
https://github.com/qqwweee/keras-yolo3/issues/171#issuecomment-522287613
